termo/app/Game.hs

```diff
@@ -1,43 +1,51 @@
 module Game where
 
-import Attempt (showAttemptNum, getAttempt, processAttempt)
+import LetterStatus (LetterStatus(..))
 import Words (getSecretWord)
-import Menu (showScore)
-
+import System.Console.ANSI (setSGR, SGR(SetColor, Reset), ColorIntensity(Vivid), Color(..), ConsoleLayer(Foreground))
+
+statusToColor :: LetterStatus -> Color
+statusToColor NoExist = Red
+statusToColor RightPlace = Green
+statusToColor WrongPlace = Yellow
+statusToColor _ = White 
+
+printStyledLetter :: (Char, LetterStatus) -> IO ()
+printStyledLetter (ch, status) = do
+    setSGR [SetColor Foreground Vivid (statusToColor status)]
+    putStr [ch]
+    setSGR [Reset]
+
+processAndPrintAttempt :: String -> String -> IO ()
+processAndPrintAttempt secretWord attempt = do
+    let statuses = zipWith (compareLetters secretWord) secretWord attempt
+    mapM_ printStyledLetter statuses
+    putStrLn ""
+
+compareLetters :: String -> Char -> Char -> (Char, LetterStatus)
+compareLetters secretWord secretLetter attemptLetter
+    | attemptLetter == secretLetter = (attemptLetter, RightPlace)
+    | attemptLetter `elem` secretWord = (attemptLetter, WrongPlace)
+    | otherwise = (attemptLetter, NoExist)
 
 runGame :: IO ()
 runGame = do
-    -- Variáveis globais do jogo
-    -- TODO: Issue 2
     secretWord <- getSecretWord
     let maxAttempts = 6
-
-    -- Loop do jogo
     loopGame secretWord maxAttempts
 
-
-
 loopGame :: String -> Int -> IO ()
 loopGame secretWord attemptNum = do
-    -- Limpando tela
-    -- putStrLn "\ESC[2J"
-
-    -- Exibindo tentativas restantes na tela
-    showAttemptNum attemptNum
-
-    -- Recebendo input do jogador
-    attempt <- getAttempt
-
-    -- Processando tentativa
-    rightAns <- processAttempt secretWord attempt
+    putStrLn ("Tentativa " ++ show attemptNum ++ ":")
+    attempt <- getLine
```

Comment:
Por que paramos de usar o `getAttemp`? Nesse método tem padronização para lowercase e evitar que inputs com caracteres minusculos se diferenciem de palavras com caracteres maiusculos.

Se for o caso para imprimir tudo em maiúscula, acho que vale transformar pra uppercase nesse método:

https://github.com/UnBParadigmas2024-2/2024.2_G3_Funcional_Termo/blob/9ee4791b9d32d39b279b0d692c19d0c104e8d2dd/termo/app/Attempt.hs#L32-L36

Mas é algo para outra issue, eu só manteria o uso nesse issue.

Reply:
Estou tendo problemas nesse método, pois verifico a palavra "TESTE" e usando esse método que transforma em minusculo ta dando erro na verificação para colocar a cor em cada letra.
Ex: "t"== "T"..
Com isso todos as palavras serão em maisculo.. decidi colocar em maiusculo mesmo.